impl/src/main/java/org/jboss/weld/resolution/QualifierInstance.java

```diff
@@ -134,7 +136,22 @@ public QualifierInstance(Class<? extends Annotation> annotationClass, Map<String
                     }
                     builder.put(method.getJavaMember().getName(), method.getJavaMember().invoke(instance));
                 } catch (IllegalAccessException | InvocationTargetException | IllegalArgumentException e) {
-                    throw ResolutionLogger.LOG.cannotCreateQualifierInstanceValues(instance, Formats.formatAsStackTraceElement(method.getJavaMember()), e);
+                    // it may happen that we are in EAR and have stored the annotation's method from different WAR class loader
+                    // an invocation will then lead to IAE, we can re-try with reflection
+                    Method[] methods;
+                    builder = ImmutableMap.builder();
+                    if (System.getSecurityManager() != null) {
+                        methods = AccessController.doPrivileged((PrivilegedAction<Method[]>)() -> annotationClass.getDeclaredMethods());
+                    } else {
+                        methods = annotationClass.getDeclaredMethods();
+                    }
+                    for (Method m : methods) {
```

Comment:
We have to filter out members annotated with `javax.enterprise.util.Nonbinding`.